### PR TITLE
feat(oidc): add set_userinfo_detail_header

### PR DESCRIPTION
**Pull request by yaule**
### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by tzssangglass** [comment]
hi @yaule , thanks for submitting the PR, can you explain the background of this PR?

File: apisix/plugins/openid-connect.lua [hunk 1]
```diff
@@ -317,6 +323,13 @@ function _M.rewrite(plugin_conf, ctx)
                 core.request.set_header(ctx, "X-Userinfo",
                     ngx_encode_base64(core.json.encode(response.user)))
             end
+            -- Add X-Userinfo-Detail header, maybe.
+            if response.user and conf.set_userinfo_detail_header then
+                for k in string.gmatch(conf.scope..' ', '(.-)'..' ')
```
Comment (starsz) [inline apisix/plugins/openid-connect.lua:328]:
```suggestion
                for k in string.gmatch(conf.scope .. ' ', '(.-)' .. ' ')
```

File: apisix/plugins/openid-connect.lua [hunk 2]
```diff
@@ -317,6 +323,13 @@ function _M.rewrite(plugin_conf, ctx)
...
+                do
+                    core.request.set_header(ctx, 'X-Userinfo-Detail-'..k,response.user[k])
```
Comment (starsz) [inline apisix/plugins/openid-connect.lua:330]:
```suggestion
                    core.request.set_header(ctx, 'X-Userinfo-Detail-' .. k, response.user[k])
```

**Comment by yaule** [comment]
> hi @yaule , thanks for submitting the PR, can you explain the background of this PR?

The service behind APISIX can directly obtain user information from the header, without using base64 to encode the data, like grafana

**Comment by tzssangglass** [comment]
Can you show the relevant docs or issue, I don't understand its usefulness.

**Comment by yaule** [comment]
> 

https://grafana.com/docs/grafana/latest/auth/auth-proxy/

**Comment by tzssangglass** [comment]
we have `set_userinfo_header` attributes in openid-connect plugin, see: https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/openid-connect.md#attributes

It looks like this feature is already available.

**Comment by yaule** [comment]
Grafana requires the header is plaintext, but X-Userinfo is base64 encoded.

**Comment by tzssangglass** [comment]
> Grafana requires the header is plaintext, but X-Userinfo is base64 encoded.

Another question, is there an introduction to `X-Userinfo-Detail-*`? Is this header specifically for Grafana?

**Comment by yaule** [comment]
Yes, the upstream services do not need base64 decoding user information, they can be obtained directly from X-userinfo-Detail-*

**Comment by tzssangglass** [comment]
> Yes, the upstream services do not need base64 decoding user information, they can be obtained directly from X-userinfo-Detail-*

My concern is that this is only grafana usage and is not a universal standard.
I would like to know if there are more common standards that support this header.

**Pull request closed without merging**